types/time.go

```diff
@@ -1272,7 +1729,8 @@ func parseTime(sc *stmtctx.StatementContext, str string, tp byte, fsp int, isFlo
 
 // ParseDatetime is a helper function wrapping ParseTime with datetime type and default fsp.
 func ParseDatetime(sc *stmtctx.StatementContext, str string) (Time, error) {
-	return ParseTime(sc, str, mysql.TypeDatetime, GetFsp(str))
+	//return ParseTime(sc, str, mysql.TypeDatetime, GetFsp(str))
+	return parseTimeNoFsp(sc, str, mysql.TypeDatetime, false)
```

Comment:
Why FSP is ignored?

Reply:
In some scene, the FSP is decided in the parse processing.I call it selfDecideFsp.